common/nodes/database-schema.xml

```diff
@@ -466,6 +466,32 @@ CREATE TABLE firmware_mapping (
 	CONSTRAINT unique_node_firmware UNIQUE (node_id, firmware_id)
 );
 
+<!-- Virtual Machines -->
+CREATE TABLE virtual_machines (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	hypervisor_id INT DEFAULT NULL,
+	node_id INT NOT NULL,
+	memory_size INT NOT NULL,
+	cpu_cores INT NOT NULL,
+	vm_delete BOOLEAN DEFAULT 0,
+	FOREIGN KEY (hypervisor_id) REFERENCES nodes(id) ON DELETE SET NULL,
+	FOREIGN KEY (node_id) REFERENCES nodes(id) ON DELETE CASCADE
+);
+
+CREATE TABLE virtual_machine_disks (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	virtual_machine_id INT DEFAULT NULL,
+	disk_name VARCHAR(128) NOT NULL,
+	disk_type VARCHAR(128) NOT NULL,
+	disk_size INT DEFAULT NULL,
+	disk_location VARCHAR(128) NOT NULL,
```

Comment:
VARCHAR is the max length of characters, right?  If so, I'd consider making these longer.  128 characters is probably usually enough, but it's just short enough that some crazy person might actually hit it and then they'd be stuck at a database level enforcement.